src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -408,7 +410,7 @@ private boolean startConnection(Socket sock, Long sid)
             dout.flush();
 
             din = new DataInputStream(
-                    new BufferedInputStream(sock.getInputStream()));
+                    new BufferedInputStream(new SocketInputStream(sock, self.tickTime * self.initLimit)));
```

Comment:
Is this the right timeout here?  Don't you want this timeout to be strictly smaller than `initLimit`? I guess that applies to most of the timeouts in this file. I suppose it's better to be consistent here.